### ISSUE TITLE
No new episodes detected in feed, even when they exist

### ISSUE THREAD

**Issue by peterzal** [gpodder/gpodder#1042]
I have a podcast that contains 50 episodes at a time, and I've been downloading them just fine until recently. I've hit 100 (i don't know if there is something special about this number causing this issue) downloads of this podcast and even though the feed itself has new episodes when I "Update podcast" it says that there are no new episodes.

I'm running the latest version from github on linux (v3.10.19)

I know this bug description is rather basic, so please let me know what additional information you'd need from me to help debug this issue further.

**Comment by auouymous** [comment]
Go to the Updating tab in preferences and increase max episodes per podcast. If the UI won't go higher, go to edit config and change `limit.episodes`.

**Comment by peterzal** [comment]
Forgot to mention that I've already done that. Its set to 2000, I have another podcast with over 1800 episodes (but this one is no longer getting new ones, so I cant verify if 'Update podcast' for this one works as expected

**Comment by auouymous** [comment]
Can you post the feed URL along with the name of an episode in the feed that isn't showing up?

**Comment by peterzal** [comment]
https://spotifeed.timdorr.com/4rOoJ6Egrf8K2IrywzwOMk

The top 2 episodes arent appearing 1654 and 1653. All of the others appear fine and I have them downloaded

**Comment by auouymous** [comment]
None of the episodes in that feed have an enclosure which is how gpodder finds media to download. The link itself is supported by youtube-dl and @tpikonen added support in #949 but it didn't pick up these episodes for me. But even if it did work, its MP3_96 format isn't supported by gpodder until #1032 is implemented.

**Comment by peterzal** [comment]
Do any of the episodes get picked up or none at all in that feed? I'm wondering if the feed creators recently changed up the feed format, because it has worked for me in the past.

**Comment by auouymous** [comment]
They probably changed the feed, it has 50 episodes and none have media.

It would be nice if gpodder showed episodes even if they lack media, at least then you could see something and trying to download a media-less episode would produce an error about no media. Would be more intuitive than an empty episode pane, and some "episodes" might be text only.

**Comment by peterzal** [comment]
Yeah, I saw that earlier this morning. Sad day. Thanks for taking the time to look into this.

**Comment by auouymous** [comment]
It appears #949 only picks up these episodes when `manage_downloads` is enabled, which makes sense, otherwise gpodder would download the link which is usually a webpage. It could set episode.url='' if link_has_media which prevents gpodder from downloading it. The youtube-dl extension would need to fallback to link when url is empty.

@tpikonen Do you see any issues with the link_has_media check some how forcing ytdl to bypass manage_downloads and always call is_supported? Also, #949 causes the play/stream item in episode context menu to say `gtk-open`.

**Comment by tpikonen** [comment]
@auouymous The idea in #949 is to parse episode.link to episode.url if there is something youtube-dl can download there. If the user has set youtube-dl.manage_downloads to False, then it cannot do that.

Generally, it could make sense to accept any RSS item and select the default media in a smarter way from enclosures and the item link using the hypothetical preferences in #1032 (and display a warning when nothing is found). There also could be an 'Open in browser' item in the context menu whenever episode.link is defined. This would then make gPodder a generic RSS reader.

The play/stream issue in CustomDL episodes is supposed to be fixed with #963. With the current master streaming is enabled for youtube and normal feeds, but not for anything else supported by youtube-dl.

**Comment by MrDaleSmith** [comment]
I think I'm getting this issue as well, with https://clarkesworldmagazine.com/feed/?rss - last item to download was "Shining Bursa and the Listening Post by Sarah Pauling (audio)" on Wed, 28 Sep 2022 20:34:39 +0000: nothing downloaded since then. I've had other feeds update since then. 

Can't see any issues with the RSS feed, or difference between the last item to download and the first to not. Have test in Doughnut and the correct episodes are downloaded without issue, and an RSS validator says the feed is ok too.

**Comment by auouymous** [comment]
@MrDaleSmith I subscribed to that feed and got two episodes published on the 30th. It couldn't be a caching issue if you can see the feed on the same computer outside of gpodder.

You could try unzipping http://borg.qzx.com/files/gpodder-debug-extension.zip to `${GPODDER_HOME}/Extensions/`, restart gpodder, enable the debug extension, right click the channel, `Clear ETag` and then update again.
